match.go

```
@@ -10,10 +10,14 @@ type Match struct {
 	pos     uint32
 	pattern uint32
 	match   []byte
+
+	// buf, set only on the first match of a batch, lets ReleaseMatches
+	// recycle the whole batch with a single pool operation.
+	buf *matchBuf
 }
 
 func newMatch(pos, pattern uint32, match []byte) *Match {
```

golangci-lint (2.12.2)

[error] func newMatch is unused

(unused)